react/week-2/lesson.md

```diff
@@ -492,55 +502,66 @@ useEffect(() => {
 And here is a more complete example:
 
 ```js
-import React, { useState, useEffect } from 'react'; // remember to import the Hook(s) you need!
+import React, { useState, useEffect } from "react"; // remember to import the Hook(s) you need!
 
 function MartianPhotoFetcher() {
-  const [marsPhotos, setMarsPhotos] = useState();
+  const [marsPhotos, setMarsPhotos] = useState({});
 
   useEffect(() => {
-    fetch(`https://api.nasa.gov/mars-photos/api/v1/rovers/curiosity/photos?earth_date=2015-6-3&api_key=DEMO_KEY`)
+    fetch(
+      `https://api.nasa.gov/mars-photos/api/v1/rovers/curiosity/photos?earth_date=2015-6-3&api_key=DEMO_KEY`
+    )
       .then(res => res.json())
       .then(data => setMarsPhotos(data));
-    });
   }, []);
 
   return (
     <div>
-      // TODO: update this example
+      {marsPhotos.photos &&
```

Comment:
I realise this is a very common & popular pattern in React, and we should have covered it in week 1, however I do think that this is simpler and easier to read:

```js
if (marsPhotos.photos) {
  return (
    <div>
      marsPhotos.photos.map(...)
    </div>
  )
} else {
  return <div />
}
```

It also points out that we're rendering a useless `<div>` if no photos are loaded.

Reply:
Might be worth returning something useful though such as a "nothing to display" message